mirdata/datasets/dagstuhl_choirset.py

```diff
@@ -0,0 +1,451 @@
+
+    Attributes:
+        mtrack_id (str): track id
+        audio_paths (list): paths to audio files
+        beat_paths (list): path to beat annotation
+
+    Cached Properties:
+        track_audio_property (str): the name of the attribute of Track which
+            returns the audio to be mixed
+        beat (annotations.BeatData): Beat annotation
+
+    """
+
+    def __init__(
+        self, mtrack_id, data_home, dataset_name, index, track_class, metadata
+    ):
+
+        super().__init__(
+            mtrack_id=mtrack_id,
+            data_home=data_home,
+            dataset_name=dataset_name,
+            index=index,
+            track_class=track_class,
+            metadata=metadata,
+        )
+
+        self.audio_paths = [
+            self.get_path(key)
+            for key in self._multitrack_paths
+            if "audio" in key
+            if self.get_path(key)
+        ]
+
+        self.beat_paths = [
+            self.get_path(key)
+            for key in self._multitrack_paths
+            if "beat" in key
+            if self.get_path(key)
+        ]
+
+    @core.cached_property
+    def track_audio_property(self):
+        return "audio_dyn"
+
+    @core.cached_property
+    def beat(self):
+        """Get beat annotation
+        Args:
+
+        Returns:
+            BeatData Object - the beat annotation
+        """
+        if not self.beat_paths:
+            return None
+
+        if len(self.beat_paths) > 1:
+            raise ValueError(
+                "Found two or more beat annotations:{}".format(self.beat_paths)
+            )
+
+        return load_beat(self.beat_paths[0])
+
+    def audio(self, mic="stm"):
+        """Get audio of the specified microphone
+        Args:
+            mic (str): Identifier of the microphone ("stm", "stereoreverb_stm", "stl", "str", "spl", or "spr")
+
+        Returns:
+            * np.ndarray - the mono audio signal
+            * float - The sample rate of the audio file
+        """
+        if mic.lower() not in ["stm", "stereoreverb_stm", "stl", "str", "spl", "spr"]:
+            raise ValueError("mic={} is invalid".format(mic))
+
+        audio_path = [s for s in self.audio_paths if mic.lower() in s.lower()]
+
+        if mic.lower() == "stm":
+            audio_path = [s for s in audio_path if "stereo_stm" in s.lower()]
+
+        if not audio_path:
+            return None
+
+        if len(audio_path) > 1:
+            raise ValueError(
+                "Found two or more microphone signals for mic={}".format(mic)
+            )
+
+        return load_audio(audio_path[0])
+
+    def to_jams(self):
+        """Jams: the track's data in jams format"""
+
+        if not self.beat_paths:
+            beat_data = None
+        else:
+            beat_data = load_beat(self.beat_paths[0])
+
+        return jams_utils.jams_converter(
+            audio_path=self.audio_paths[0], beat_data=[(beat_data, "beat")]
+        )
```

Comment:
Our dataset has several kinds of microphone signals. Some of them only belong to the Track class (lrx, dyn, hsm) and others only belong to the MultiTrack class (stm, str, stl, spr, spl, stereoreverb). We are not providing a function to artificially mix tracks, because we have the microphone signals from the multitrack class.
In this light, we think having two different audio() functions makes sense, because they load different audio files.